benchmark/scenarios/test/test_write_request_load.py

```diff
@@ -167,6 +168,39 @@ def assert_created(returned_datasets):
         d2.addCallback(assert_created)
         s.stop()
 
+    @capture_logging(None)
+    def test_cleanup_deletes_dataset(self, _logger):
+        """
+        ``write_request_load_scenario`` deletes the dataset created by
+        the setup when the scenario is stopped.
+        """
```

Comment:
I could change the naming in this test, but refactoring the whole file may be out of scope for this issue. Please, feel free to raise a issue if you think it needs a bit of refactoring - it may not be a bad idea, as they could be simpler and better structured